src/Containers/FileUploadV2/useInformativePanels.ts

```diff
@@ -0,0 +1,206 @@
+            panels: [],
+            makeItDisappear: [],
+            startWorkers: [],
+        });
+
+    /**
+     * set end state
+     */
+    const prepareForEndInformativePanel = useCallback(
+        (operationState: OperationState, informativePanel: IPanel): void => {
+            setInformativePanels((prev) => ({
+                ...prev,
+                panels: prev.panels.map((panel) => {
+                    const {name}=panel;
+                    if (name === informativePanel.name)
+                        return {
+                            ...panel,
+                            operationState,
+                        };
+                    return panel;
+                }),
+                makeItDisappear: [
+                    ...prev.makeItDisappear,
+                    informativePanel.name,
+                ],
+            }));
+        },
+        [],
+    );
+
+    /**
+     * terminate worker and set state of informative panel to success or failure and
+     * send order to remove informative panel in the future. also do whatever user
+     * wants to do with the file read in case of success
+     */
+    const onWorkerMessage = useCallback(
+        (e: any) => {
+            const { base64StringFile, name } = e.data;
+            if (base64StringFile === undefined) {
+                return;
+            }
+            const informativePanel = informativePanels.panels.find(
+                (panel) => panel.name === name,
+            );
+            if (informativePanel) {
+                let operationState:OperationState=OperationState.isUnknown;
+                if (
+                    base64StringFile === NO_BASE64STRINGFILE ||
+                    isTestIsFailure
+                ) {
+                    operationState=OperationState.isFailure;
+                } else {
+                    onFile(base64StringFile);
+                    operationState=OperationState.isSuccess;
+                }
+                prepareForEndInformativePanel(operationState,informativePanel);
+            }
+        },
+        [
+            informativePanels.panels,
+            isTestIsFailure,
+            onFile,
+            prepareForEndInformativePanel,
+        ],
+    );
+
+    // start workers after files have been droped and array of informative panels
+    // are loaded
+    useEffect(() => {
+        if (informativePanels.startWorkers.length) {
+            informativePanels.startWorkers.forEach((name) => {
+                const informativePanel = informativePanels.panels.find(
+                    (panel) => panel.name === name,
+                );
+                if (informativePanel && informativePanel.worker) {
+                    const {worker:informativePanelWorker}=informativePanel;
+                    informativePanelWorker.onmessage = onWorkerMessage;
+                    informativePanelWorker.postMessage({
+                        file: informativePanel.file,
```

Comment:
Restructure with above

Reply:
what do you mean, this const {worker:{onmessage,postMessage}=informativePanel? 
this doesn't work, I've already tried.

Reply:
const {file, worker:informativePanelWorker}=informativePanel;